extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -538,13 +538,13 @@ export async function promptConnectionStringPasswordAndUpdateConnectionString(co
 				connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, true, false);
 			}
 		}
-		// set connection string to not include the password if connection info does not include password, or user chooses to not include password, or authentication type is not sql login
 		let userPassword: string | undefined;
-		if (includePassword !== constants.yesString) {
+		// set connection string to not include the password if connection info does not include password, 
+		// or user chooses to not include password, or authentication type is not SQL login
+		if (includePassword !== constants.yesString || !connectionInfo.password || connectionInfo.authenticationType !== 'SqlLogin') {
 			connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, false, false);
-
-			// Ask user to enter password if auth type is sql login and password is not saved
-			if (connectionInfo.authenticationType === 'SqlLogin' && connectionInfo.password) {
+			if (!includePassword && connectionInfo.authenticationType === 'SqlLogin') {
```

Comment:
Shouldn't this be `!connectionInfo.password && connectionInfo.authenticationType === 'SqlLogin'`? 

I'm not even sure it makes sense for this to be in this if block though - this seems like it should only be handling the case where the user doesn't want to include the password (chose no or cancelled out).

The only time we should be prompting them for the password is when they chose to keep the password but we didn't have one in the connection info. So to keep this cleaner I'd move it into an else if.

Reply:
1. Scenario 1: If User uses a non-SQL auth then we get connection string with no password (do not show warning or include password prompt).
2. Scenario 2: If User uses SQL auth and connection info password is saved then ask to include the password:
a. If user selects yes, then add password to connection string
b. If user selects no, then don't add a password to the connection string and warn the user that they will have to include password (do not prompt to manually enter it)
3. Scenario 3: If user uses SQL auth and the connection info password is NOT saved then ask user to manually enter a password or warn them:
a. If user enters password, then added password to connection string
b. If user cancels, then don't add password to connection string and warn user that they will have to include password

Reply:
If I put this in an `else if` then for Scenario 3.b would not warn the user. That is why we need to have it in a separate block.

Reply:
Also, we need to get the `connectionString` on line 545 (above) to be able to replace the connection string after the user enters the password replacement. So if I have that in an `else if` for line 530 (I assume you meant) then we would not have the connectionString set properly by then.

Reply:
I think this function would greatly benefit from some slight refactoring to better organize the code. Basically have it be 

```
if (<scenario 1 checks>) {
   // handle scenario 1 here
} else if (<scenario 2 checks) {
  // handle scenario 2 here
} else {
   // scenario 3 
}
```

This would make it a lot easier to follow/verify and I don't really see any benefit from the nested checks you have now since each scenario is pretty much separate from each other.

Reply:
The main thing that I wanted to keep was the getConnectionString where we set includePassword parameter to `false` which will be used for scenario 1, 2.b, and scenario 3. 

I can have each one have the getConnectionString request but not sure if that is cleaner.

Reply:
I definitely think it'd be cleaner (repeating 1 line of code in a couple places isn't a huge deal I think) but it's also not blocking so I'm fine if you want to hold off and do that later.

Reply:
<img width="762" alt="Screen Shot 2022-05-18 at 5 23 51 PM" src="https://user-images.githubusercontent.com/23587151/169177125-6d368111-ead0-434b-bed8-b874273af265.png">